CMakeLists.txt

```diff
@@ -1,52 +1,125 @@
-cmake_minimum_required(VERSION 3.11)
+# Detray library, part of the ACTS project (R&D line)
+#
+# (c) 2021 CERN for the benefit of the ACTS project
+#
+# Mozilla Public License Version 2.0
 
-project(detray VERSION 0.1 LANGUAGES CXX)
+# Set up the project.
+cmake_minimum_required( VERSION 3.11 )
+project( detray VERSION 0.1 LANGUAGES CXX )
 
-option(CMAKE_WARN_DEPRECATED "Not warning for deprecation in cmake" Off)
-enable_testing()
+# CMake include(s).
+include( CMakeDependentOption )
+include( GNUInstallDirs )
 
-include(CMakeDependentOption)
+# Explicitly set the output directory for the binaries. Such that if this
+# project is included by another project, the main project's configuration would
+# win out.
+set( CMAKE_RUNTIME_OUTPUT_DIRECTORY
+   "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}" CACHE PATH
+   "Directory for the built binaries" )
+set( CMAKE_LIBRARY_OUTPUT_DIRECTORY
+   "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}" CACHE PATH
+   "Directory for the built libraries" )
+set( CMAKE_ARCHIVE_OUTPUT_DIRECTORY
+   "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}" CACHE PATH
+   "Directory for the built static libraries" )
 
 option(DETRAY_ARRAY_PLUGIN "Build std::array math plugin" On)
```

Comment:
How is this linked to the algebra-plugins now?

Reply:
Right now it's not. The default/starting values of the `ALGEBRA_` cache variables could be set based on the `DETRAY_` flags. For some flag-combinations it's relatively easy to establish such links. For others it's a bit more tricky.

Reply:
It just happened the other day that I was trying to set a detray flag, which had no effect after all, and I ended up compiling with double precision and got confused why the benchmarks were so terrible :)